### PR TITLE
Caregivers Backend Integration

### PR DESCRIPTION
**Pull request by LinnJS**
## Description
Integrate frontend with backend

## Screenshots

Screen Shot 2020-05-29 at 2 02 31 AM

Matches [Full Schema](https://github.com/department-of-veterans-affairs/vets-json-schema/blob/master/src/schemas/10-10CG/schema.js)

## Acceptance criteria
- [x] Form submits and behaves as expected 

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/caregivers/helpers.js [hunk 1]
```diff
@@ -19,7 +20,97 @@ export const medicalCenterLabels = Object.keys(caregiverFacilities).reduce(
...
+        case 'primary':
+          return 'primaryCaregiver';
+        case 'secondaryOne':
+          return 'secondaryCaregiverOne';
+        case 'secondaryTwo':
+          return 'secondaryCaregiverTwo';
+        default:
+          return null;
+      }
+    };
```
Comment (bradl3yC) [inline src/applications/caregivers/helpers.js:57]: You can make this an object
```
const chapterNames = Object.freeze({
  'veteran': 'veteran',
  'primary': 'primaryCaregiver',
  'secondaryOne': 'secondaryCaregiverOne',
  'secondaryTwo': 'secondaryCaregiverTwo',
}
```
then when you need it just use:
```
chapterNames[key]
```

Comment (LinnJS) [reply src/applications/caregivers/helpers.js:57]: I am going to have to look into this at another time, I am having issues null checking it, one of the nice parts about the switch is that default statement which helps reject undefined. Will look into this feedback and see if I can get it working with less code down the road.

**Review by bradl3yC — APPROVED** [review]

**Review by maharielrosario — APPROVED** [review]

File: src/applications/caregivers/helpers.js [hunk 2]
```diff
@@ -19,7 +20,97 @@ export const medicalCenterLabels = Object.keys(caregiverFacilities).reduce(
...
+const medicalCentersByState = _.mapValues(
   val => val.map(center => center.code),
   caregiverFacilities,
 );
+
+// transforms forData to match fullSchema structure for backend submission
+const submitTransform = (formConfig, form) => {
+  // checks for optional chapters using ssnOrTin
+  const hasSecondaryOne =
+    form.data.secondaryOneSsnOrTin === undefined ? null : 'secondaryOne';
```
Comment (Mottie) [inline src/applications/caregivers/helpers.js:32]: We can use [null coalescing operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) now, I think... so this becomes

```js
const hasSecondaryOne = form.data.secondaryOneSsnOrTin = null ?? 'secondaryOne';
```

File: src/applications/caregivers/helpers.js (hunk 1 above)
Comment (Mottie) [reply src/applications/caregivers/helpers.js:57]: Oh yeah, LOL I just recommended this same thing LOL

**Review by Mottie — APPROVED** [review]

**Pull request merged**